lisa/tools/python.py

```diff
@@ -89,3 +90,66 @@ def exists_package(self, package_name: str) -> bool:
     def uninstall_package(self, package_name: str) -> bool:
         result = self.run(f"uninstall {package_name} -y", force_run=True, sudo=True)
         return result.exit_code == 0
+
+
+class PythonVenv(Python):
+    @property
+    def command(self) -> str:
+        return f"{self.get_venv_path()}/bin/{super().command}"
+
+    def _check_exists(self) -> bool:
+        # _super = type(super())
+        # assert isinstance(super(), Python)
+        return self.node.execute("python3 -m venv --help", shell=True).exit_code == 0
+        # return self.node.execute("python3 -m venv --help").exit_code == 0
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        super()._initialize(*args, **kwargs)
+
+    def _install(self) -> bool:
+        super()._install()
+        if isinstance(self.node.os, Posix):
+            self.node.os.install_packages("python3-venv")
+        return self._check_exists()
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Python]
+
+    # if get_venv_path is invoked before create_venv, it will create
+    # a venv in the node working path
+    def get_venv_path(self) -> PurePath:
+        if not hasattr(self, "_venv_path"):
+            self._venv_path = self.create_venv()
+        return self._venv_path
+
+    def create_venv(self, venv_path: str = "", sudo: bool = False) -> PurePath:
```

Comment:
@squirrelsc  Should virtual env be created in _initialize()? But this would mean it user cannot pass the installation path

Reply:
Why do we need venv, instead of direct install in the current Python?

Reply:
Observed this error in GPU test suite for Ubuntu 2310

2023-12-11 03:50:58.204[139645954287168][DEBUG] lisa.env[generated_0].node[0].cmd[7803] cmd: ['pip3', 'install', '-q', 'torch'], cwd: None, shell: False, sudo: False, nohup: False, posix: True, remote: True, encoding: utf-8
2023-12-11 03:50:59.501[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout error: externally-managed-environment
2023-12-11 03:50:59.501[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout × This environment is externally managed
2023-12-11 03:50:59.502[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout ╰─> To install Python packages system-wide, try apt install
2023-12-11 03:50:59.503[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     python3-xyz, where xyz is the package you are trying to
2023-12-11 03:50:59.503[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     install.
2023-12-11 03:50:59.505[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     If you wish to install a non-Debian-packaged Python package,
2023-12-11 03:50:59.507[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     create a virtual environment using python3 -m venv path/to/venv.
2023-12-11 03:50:59.507[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
2023-12-11 03:50:59.508[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     sure you have python3-full installed.
2023-12-11 03:50:59.509[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     If you wish to install a non-Debian packaged Python application,
2023-12-11 03:50:59.510[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     it may be easiest to use pipx install xyz, which will manage a
2023-12-11 03:50:59.511[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     virtual environment for you. Make sure you have pipx installed.
2023-12-11 03:50:59.512[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout     See /usr/share/doc/python3.11/README.venv for more information.
2023-12-11 03:50:59.514[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
2023-12-11 03:50:59.515[139646265976384][DEBUG] lisa.env[generated_0].node[0].cmd[7803].stdout hint: See PEP 668 for the detailed specification.

Reply:
I thought it is better to have cases install packages in venv to avoid conflicts. We can also remove the packages after the testcase to clean up space.

Reply:
Sorry, I did not understand. Are you suggesting we can do that using current python tool?
Or specifying path to this venv tool in the PR?

Reply:
You can do it when creating the venv tool. So you can pass an argument, and receiving it when calling `__init__`.

Reply:
@squirrelsc , I have made the changes, can you please review